cmd/tasks/src/lib.rs

```diff
@@ -150,13 +149,23 @@ struct TasksArgs {
     task: Option<String>,
 }
 
-fn print_regs(regs: &BTreeMap<ARMRegister, u32>, additional: bool) {
+fn print_regs(
+    regs: &BTreeMap<Register, u32>,
+    additional: bool,
+    hubris: &HubrisArchive,
+) {
     let bar = if additional { "|" } else { " " };
 
     print!("   |\n   +--->");
 
     for r in 0..=16 {
-        let reg = ARMRegister::from_usize(r).unwrap();
+        if let Some(goblin::elf::header::EM_RISCV) = hubris.arch {
+            // the riscv zero register is not saved
```

Comment:
Is this function reading from the saved state on stack?

Reply:
It is really just printing whatever registers that get passed in.  In this case it is printing the general purpose registers or if the task is in a syscall, the registers `SavedState` struct within the kernel

Reply:
I see.